### ISSUE TITLE
Suggest needs a more flexible inputValueRenderer 

### ISSUE THREAD

**Issue by tnrich** [palantir/blueprint#3556]
I think it would be great if the Suggest component (and perhaps InputGroup's in general) could optionally render something other than a string as their value. For example, a color Suggest would be very useful to have the input look more like: 

![image](https://user-images.githubusercontent.com/2730609/57737568-f9b77a80-7660-11e9-9451-f485560f1cc3.png)

Any thoughts on this @giladgray ? 

Thanks!

**Comment by giladgray** [comment]
@tnrich I agree with this request but there's a pretty significant hurdle here: `Suggest` renders an `input` which only accepts string values. You can, however, use `inputProps.icon` etc to put the colored square.

**Comment by tnrich** [comment]
I think it should probably render a modified TagInput in that case. I'm going to basically modify the MultiSelect to turn it into a single select for my needs because the Tag value can be much more easily customized.

**Comment by giladgray** [comment]
@tnrich please share how that goes, I'm sure the results will be interesting.
